src/ui/UiImageUploader.tsx

```diff
@@ -0,0 +1,76 @@
+import { Avatar, Input, type InputProps, Stack, type StackProps, Typography } from '@mui/material'
+import { ChangeEvent, forwardRef, useCallback } from 'react'
+
+import { useThemeMode } from '@/hooks'
+import UiIcon from '@/ui/UiIcon'
+
+interface Props extends Omit<InputProps, 'value' | 'onChange'> {
+  label?: string
+  stackProps?: StackProps
+  value?: File
+  onChange?: (value: File) => void
+}
+
+const AVATAR_SIZE = 19
+
+const UiImageUploader = forwardRef<HTMLInputElement, Props>(
```

Comment:
it does not upload anything, so I'd name it `UiImageInput` or `UiImageField` or even `UiFileField`